front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -243,116 +86,131 @@ export const usePathfinding = (
-        const stopType = i === pathSteps.length - 1 && !step.stopFor ? undefined : step.stopType;
-
-        return {
-          ...step,
-          positionOnPath: pathResult.path_item_positions[i],
-          stopFor,
-          stopType,
-          theoreticalMargin,
-          ...(correspondingOp && {
-            name: correspondingOp.name,
-            uic: correspondingOp.uic,
-            secondary_code: correspondingOp.ch,
-            kp: correspondingOp.kp,
-            coordinates: correspondingOp.coordinates,
-          }),
-        };
-      });
+      const theoreticalMargin = i === 0 ? step.theoreticalMargin || '0%' : step.theoreticalMargin;
 
-      if (!isEmptyArray(powerRestrictions)) {
-        dispatch(
-          setWarning({
-            title: t('warningMessages.pathfindingChange'),
-            text: t('warningMessages.powerRestrictionsReset'),
-          })
-        );
-      }
-      dispatch(updatePathSteps(updatedPathSteps));
+      const stopFor = i === pathStepsInput.length - 1 && !step.stopFor ? '0' : step.stopFor;
+      const stopType = i === pathStepsInput.length - 1 && !step.stopFor ? undefined : step.stopType;
+
+      return {
+        ...step,
+        positionOnPath: pathResult.path_item_positions[i],
+        stopFor,
+        stopType,
+        theoreticalMargin,
+        ...(correspondingOp && {
+          name: correspondingOp.name,
+          uic: correspondingOp.uic,
+          secondary_code: correspondingOp.ch,
+          kp: correspondingOp.kp,
+          coordinates: correspondingOp.coordinates,
+        }),
+      };
+    });
 
-      const allWaypoints = upsertPathStepsInOPs(
-        suggestedOperationalPoints,
-        compact(updatedPathSteps)
+    if (!isEmptyArray(powerRestrictions)) {
+      dispatch(
+        setWarning({
+          title: t('warningMessages.pathfindingChange'),
+          text: t('warningMessages.powerRestrictionsReset'),
+        })
       );
+    }
+    dispatch(updatePathSteps(updatedPathSteps));
 
-      setPathProperties?.({
-        electrifications,
-        geometry,
-        suggestedOperationalPoints,
-        allWaypoints,
-        length: pathResult.length,
-        trackSectionRanges: pathResult.track_section_ranges,
-        incompatibleConstraints,
-      });
-    };
+    const allWaypoints = upsertPathStepsInOPs(
+      suggestedOperationalPoints,
+      compact(updatedPathSteps)
+    );
+
+    setPathProperties({
+      electrifications,
+      geometry,
+      suggestedOperationalPoints,
+      allWaypoints,
+      length: pathResult.length,
+      trackSectionRanges: pathResult.track_section_ranges,
+      incompatibleConstraints,
+    });
+  };
+
+  const launchPathfinding = useCallback(
+    async (steps: (PathStep | null)[]) => {
+      dispatch(replaceItinerary(steps));
+      setPathProperties(undefined);
 
-    const startPathFinding = async () => {
-      if (pathfindingState.running) {
+      if (steps.some((step) => step === null)) {
```

Comment:
If you replace this line with this, typescript will understand that steps can't have any `null` value and you can avoid all null checks in the rest of the function and in `getPathfindingQuery`

```suggestion
      if (!steps.every((step) => step !== null)) {
```

Reply:
Hum I still have the ts error, how did you do ?

Reply:
hmmm 

<img width="677" alt="Capture d’écran 2024-12-12 à 17 31 05" src="https://github.com/user-attachments/assets/86659252-dc84-4743-b987-e396cda176ba" />

Reply:
Ok so looks like I have the typescript 5.5+ on my setup (maybe because of vscode) which fixes this behavior but the app is still in 5.3.

We should keep this in mind when bumping typescript version

Reply:
issue is here : https://github.com/OpenRailAssociation/osrd/issues/10081